spec/requests/shared_examples/internal_policy_dependant_request.rb

```diff
@@ -0,0 +1,15 @@
+RSpec.shared_examples "an InternalPolicy dependant request" do |resource|
+  let(:user) { create(:user) }
+
+  before do
+    user.add_role(:single_resource_admin, resource)
+    sign_in user
+    allow(InternalPolicy).to receive(:new).and_call_original
+  end
+
+  it "responds with 200 OK" do
```

Comment:
Great. Could we also test the case when the user is not authorized to perform an aciton?

Reply:
Done!